src/Components/MediaChannelComponent.ts

```diff
@@ -0,0 +1,176 @@
+import {Component} from "../Component";
+import {GuildMember, Message, MessageReaction, TextChannel, User, VoiceState} from "discord.js";
+import {ComponentNames} from "../Constants/ComponentNames";
+import {isAdmin} from "../HelperFunctions";
+import {DJMTGuild} from "../DJMTGuild";
+import {ComponentCommands} from "../Constants/ComponentCommands";
+
+/**
+ * Declare data you want to save in JSON here. This interface is used for getSaveData and
+ * afterLoadJSON, as it tells Typescript what data you're expecting to write and load.
+ */
+interface MediaComponentSave {
+    channels: Array<string>;
+}
+
+/**
+ * A component to declare media only channels and remove any messages without attachments in said channels.
+ */
+export class MediaChannelComponent extends Component<MediaComponentSave> {
+
+    // MANDATORY: Define a name in ComponentNames.ts and place it here.
+    name: ComponentNames = ComponentNames.MEDIA_CHANNEL;
+    channelsArray: Array<TextChannel> = new Array<TextChannel>();
```

Comment:
```suggestion
    channelsArray:TextChannel[] = [];
```
This is equivalent as well